server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -213,67 +207,37 @@ public int hashCode() {
     public static final class TimestampField implements Writeable, ToXContentObject {
 
         static ParseField NAME_FIELD = new ParseField("name");
-        static ParseField FIELD_MAPPING_FIELD = new ParseField("mapping");
 
         @SuppressWarnings("unchecked")
         private static final ConstructingObjectParser<TimestampField, Void> PARSER = new ConstructingObjectParser<>(
             "timestamp_field",
-            args -> new TimestampField((String) args[0], (Map<String, Object>) args[1])
+            args -> new TimestampField((String) args[0])
         );
 
         static {
             PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME_FIELD);
-            PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> p.mapOrdered(), FIELD_MAPPING_FIELD);
         }
 
         private final String name;
-        private final Map<String, Object> fieldMapping;
-
-        public TimestampField(String name, Map<String, Object> fieldMapping) {
-            assert fieldMapping.containsKey("type") : "no type defined for mapping of timestamp_field";
-            assert ALLOWED_TIMESTAMPFIELD_TYPES.contains(fieldMapping.get("type")) :
-                "invalid type defined for mapping of timestamp_field";
 
+        public TimestampField(String name) {
+            assert "@timestamp".equals(name) : "unexpected timestamp field [" + name + "]";
```

Comment:
I think we should transform this into an actual error, that way if we end up using it for the HLRC it doesn't fail to throw because assertions are disabled

Reply:
I think HLRC uses its own `DataStream` pojo. I will change this from an assertion to an actual error.